jobsdb/jobsdb.go

```diff
@@ -3124,11 +3124,17 @@ func (jd *HandleT) getUnprocessed(ctx context.Context, params GetQueryParamsT) (
 	if jd.dsLimit != nil {
 		dsLimit = *jd.dsLimit
 	}
-	for _, ds := range dsList {
+	for _, dsRange := range dsRangeList {
+		if params.AfterJobID != nil {
+			if dsRange.maxJobID < *params.AfterJobID {
```

Comment:
AFAIK the last dataset doesn't have a `maxJobID`

Reply:
true
please take a look again